monai/apps/detection/transforms/dictionary.py

```diff
@@ -687,10 +697,81 @@ def inverse(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, Nd
 
 
+class ClipBoxToImaged(MapTransform, InvertibleTransform):
+    """
+    Dictionary-based wrapper of :py:class:`monai.apps.detection.transforms.array.ClipBoxToImage`.
+
+    Clip the bounding boxes and the associated labels/scores to makes sure they are within the image.
+    There might be multiple keys of labels/scores associated with one key of boxes.
+
+    Args:
+        box_keys: The single key to pick box data for transformation. The box mode is assumed to be ``StandardMode``.
+        label_keys: Keys that represents the lables corresponding to the ``box_keys``.
+        box_ref_image_keys: The single key that represents the reference image
+            to which ``box_keys`` and ``label_keys`` are attached.
+        remove_empty: whether to remove the boxes that are actually empty
+        allow_missing_keys: don't raise exception if key is missing.
+
+    Example:
+        .. code-block:: python
+
+            ClipBoxToImaged(
+                box_keys="boxes", box_ref_image_keys="image", label_keys=["labels", "scores"], remove_empty=True
+            )
+    """
+
+    def __init__(
+        self,
+        box_keys: KeysCollection,
+        label_keys: KeysCollection,
+        box_ref_image_keys: KeysCollection,
+        remove_empty: bool = True,
+        allow_missing_keys: bool = False,
+    ) -> None:
+        box_keys_tuple = ensure_tuple(box_keys)
+        if len(box_keys_tuple) != 1:
+            raise ValueError(
+                "Please provide a single key for box_keys.\
+                All label_keys are attached to this box_keys."
+            )
+        box_ref_image_keys_tuple = ensure_tuple(box_ref_image_keys)
+        if len(box_ref_image_keys_tuple) != 1:
+            raise ValueError(
+                "Please provide a single key for box_ref_image_keys.\
+                All box_keys and label_keys are attached to this box_ref_image_keys."
+            )
+        self.label_keys = ensure_tuple(label_keys)
+        super().__init__(box_keys_tuple + self.label_keys, allow_missing_keys)
+
+        self.box_keys = box_keys_tuple[0]
+        self.box_ref_image_keys = box_ref_image_keys_tuple[0]
+        self.clipper = ClipBoxToImage(remove_empty=remove_empty)
+
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
+        d = dict(data)
+        spatial_size = d[self.box_ref_image_keys].shape[1:]
+        self.push_transform(d, self.box_keys)
+        labels = [d[label_key] for label_key in self.label_keys]
+        d[self.box_keys], clipped_labels = self.clipper(d[self.box_keys], labels, spatial_size)
+
+        for label_key, clipped_labels_i in zip(self.label_keys, clipped_labels):
+            self.push_transform(d, label_key)
+            d[label_key] = clipped_labels_i
+        return d
+
+    def inverse(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
+        d = deepcopy(dict(data))
+        # clipped boxes cannot be restored
```

Comment:
If it doesn't support inverse, please don't inherit from `InvertibleTransform`.

Thanks.